app/components/onboarding/AddNewSite.tsx

```diff
@@ -12,47 +14,99 @@ import {
 } from '@mui/material';
 import { palettes } from '@/app/[lang]/theme/palettes';
 import { Add, ArrowBack, PlaceOutlined } from '@mui/icons-material';
-import AddNewSiteDialog from '@/app/components/onboarding/AddNewSiteDialog';
 import TitleBar from '@/app/components/onboarding/TitleBar';
 import type {
   SiteData,
   UserFormData,
 } from '@/app/components/onboarding/OnboardingDashboard';
 import type { Dict } from '@/app/[lang]/dictionaries';
 import { cropTypes } from '@/app/[lang]/dictionaries';
+import { LatLng, LatLngExpression } from 'leaflet';
+import dynamic from 'next/dynamic';
+
+const AddNewSitePosition = dynamic(
+  () => import('@/app/components/onboarding/AddNewSitePosition'),
+  { ssr: false }
+);
 
 interface OnboardingAddNewSiteProps {
   dict: Dict;
   setOnboardingStep: (value: number) => void;
   values: UserFormData;
   setValues: (values: UserFormData) => void;
+  siteToView: number;
+  setSiteToView: (value: number) => void;
 }
 
 const initialValues: SiteData = {
   name: '',
   type: '',
-  location: '',
+  position: '',
+  radius: '0',
 };
 
 const initialErrors: SiteData = {
   name: '',
   type: '',
-  location: '',
+  position: '',
+  radius: '',
 };
 
 const AddNewSite = ({
   dict,
   setOnboardingStep,
   values,
   setValues,
+  siteToView,
+  setSiteToView,
 }: OnboardingAddNewSiteProps) => {
   const [siteValues, setSiteValues] = useState<SiteData>(initialValues);
   const [errors, setErrors] = useState<SiteData>(initialErrors);
-  const [openAddSite, setOpenAddSite] = useState(false);
   const [submitAttempted, setSubmitAttempted] = useState<boolean>(false);
+  const [position, setPosition] = useState<LatLngExpression | null>(null);
+  const [radius, setRadius] = useState<number>(0);
+  const [openAddSite, setOpenAddSite] = useState(false);
+  const [positionInfo, setPositionInfo] = useState('');
+
+  useEffect(() => {
+    if (siteToView !== -1) {
+      setSiteValues(values.sites[siteToView]);
+      setRadius(Number(values.sites[siteToView].radius));
+    }
+  }, [siteToView, values.sites]);
+
+  useEffect(() => {
+    if (siteValues.position) {
+      const getLocation = async () => {
+        const response = await fetch(
+          `/api/geocoding/reverse?lat=${
+            JSON.parse(siteValues.position).lat
+          }&lon=${JSON.parse(siteValues.position).lng}`,
+          {
+            method: 'GET',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+          }
+        );
+        const data = await response
+          .json()
+          .then((res) => res.data.features[0].properties);
+
+        setPositionInfo(data.city + ', ' + data.country);
```

Comment:
Somewhat subjective, but I think this is more readable with string interpolation:

```javascript
setPositionInfo(`${data.city}, ${data.country}`);
```